src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -608,35 +606,48 @@ trait ResourceRoutes extends SamUserDirectives with SecurityDirectives with SamM
       "includePublic" ? false,
       "format".as[String] ? "hierarchical"
     ) { (resourceTypes: Option[String], policies: Option[String], roles: Option[String], actions: Option[String], includePublic: Boolean, format: String) =>
-      format match {
-        case "flat" =>
-          complete {
-            resourceService
-              .listResourcesFlat(
-                samUser,
-                resourceTypes.map(_.split(",").map(ResourceTypeName(_)).toSet).getOrElse(Set.empty),
-                policies.map(_.split(",").map(AccessPolicyName(_)).toSet).getOrElse(Set.empty),
-                roles.map(_.split(",").map(ResourceRoleName(_)).toSet).getOrElse(Set.empty),
-                actions.map(_.split(",").map(ResourceAction(_)).toSet).getOrElse(Set.empty),
-                includePublic,
-                samRequestContext
-              )
-              .map(StatusCodes.OK -> _)
-          }
-        case "hierarchical" =>
-          complete {
-            resourceService
-              .listResourcesHierarchical(
-                samUser,
-                resourceTypes.map(_.split(",").map(ResourceTypeName(_)).toSet).getOrElse(Set.empty),
-                policies.map(_.split(",").map(AccessPolicyName(_)).toSet).getOrElse(Set.empty),
-                roles.map(_.split(",").map(ResourceRoleName(_)).toSet).getOrElse(Set.empty),
-                actions.map(_.split(",").map(ResourceAction(_)).toSet).getOrElse(Set.empty),
-                includePublic,
-                samRequestContext
-              )
-              .map(StatusCodes.OK -> _)
-          }
+      val resourceTypeNames = resourceTypes.map(_.split(",").map(ResourceTypeName(_)).toSet).getOrElse(Set.empty)
+      val accessPolicyNames = policies.map(_.split(",").map(AccessPolicyName(_)).toSet).getOrElse(Set.empty)
+      val resourceRoleNames = roles.map(_.split(",").map(ResourceRoleName(_)).toSet).getOrElse(Set.empty)
+      val resourceActions = actions.map(_.split(",").map(ResourceAction(_)).toSet).getOrElse(Set.empty)
+      getWithTelemetry(
+        samRequestContext,
+        resourceTypeNameParams(resourceTypeNames) ++
+          policyNameParams(accessPolicyNames) ++
+          roleNameParams(resourceRoleNames) ++
+          actionNameParams(resourceActions) :+
+          ("includePublic" -> BooleanValue(includePublic)): _*
```

Comment:
This is not doing what you want. The *WithTelemetry directives are for parameters in the URI path, not query parameters. The path in this case will always be `/api/resources/v2`.